spring-kafka/src/main/java/org/springframework/kafka/support/serializer/JsonDeserializer.java

```diff
@@ -365,10 +376,14 @@ public void setTypeFunction(BiFunction<byte[], Headers, JavaType> typeFunction)
 	 */
 	public void setTypeResolver(JsonTypeResolver typeResolver) {
 		this.typeResolver = typeResolver;
+		this.configured = true;
 	}
 
 	@Override
 	public void configure(Map<String, ?> configs, boolean isKey) {
+		if (this.configured) {
+			return;
```

Comment:
May be better to throw an exception (or at least WARN) to let end-users to know that they are wrong?

Reply:
We can't do that; because the supplier is called each time a new consumer is created. The default suppliers return the same instance each time.

I could add a DEBUG log; WDYT?

Reply:
Got it. Then WARN anyway: people have to know that they are doing something wrong. 
With debug it may slip out and then we will get some issue that they config doesn't work this or that way.

Reply:
Hmmm - warn would be too much noise - maybe I should use 2 flags, one for setters (and throw exception) and one for `configAlreadyDone`?

Reply:
Or, maybe just close this as won't fix?

Reply:
It will be, but it will say you that there is something wrong with the app. Then you go to docs and JavaDocs and you see that you can't combine setters configuration with properties one.
Isn't the the reason why we use that `Assert` all around?

Reply:
My concern is a single shared version is used and if we call `configure()` multiple times, even though the complete operation is idempotent, the configuration will be in an indeterminate state while `configure()` is running, and another thread might be using it to deserialize.
I therefore only want to run `configure()` once.

Hence my suggestion to use 2 flags instead of 1.

Reply:
Working on polishing it...